topics/transcriptomics/tutorials/pinpoint-key-pathways-with-heinz/tutorial.md

```diff
@@ -0,0 +1,404 @@
+
+Similar to **Fit a BUM model**, we also need to prepare the input data for the tool
+**Calculate a Heinz score**.
+
+> ### {% icon question %} Question
+>
+> What is the requirement of the input data format for **Calculate a Heinz score**?
+>
+> > ### {% icon solution %} Solution
+> >
+> > In the user interface of the tool "Calculate a Heinz score", we see that "A node file with p-values" is needed.
+> > It should contain two columns delimited by tab, one is KO ID; the other, p-value.
+> >
+> {: .solution}
+{: .question}
+
+> ### {% icon hands_on %} Hands-on: extract geneID and p-values from DESeq2 output
+>
+> - **cut** columns from a table {% icon tool %} with the following parameters
+>   - *"Cut columns"*: `c1,c6`
+>   - *"Delimited by"*: `TAB`
+>   - *"From"*: the output of **DESeq2**
+>
+{: .hands_on}
+
+> ### {% icon hands_on %} Hands-on: calculate Heinz scores
+>
+> - **Calculate a Heinz score** {% icon tool %} with the following parameters
+>   - *"A node file with p-values"*: the output of the previous **cut** step
+>   - *"FDR value"*: to 0.11
+>   - *"Choose your input type for BUM parameters"*: `The output file of BUM model`
+>   - *"Output file of BUM model as input: lambda on the first line and alpha, the second"*: the output of **Fit a BUM model**
+{: .hands_on}
+
+## Run Heinz: pinpoint the optimal subnetwork
+
+After getting Heinz scores, let's run Heinz program to find the optimal subnetwork from the background network which we mentioned earlier.
+
+> ### {% icon hands_on %} Hands-on: pinpoint the optimal subnetwork
+>
+> - **Identify optimal scoring subnetwork** {% icon tool %} with the following parameters
+>   - *"File containing Heinz scores"*: the output of **Calculate a Heinz score**
+>   - *"Edge file"*: the edge file uploaded
+>
+{: .hands_on}
+
+It usually takes a few minutes to get the result, but mind you, for some tasks, it might take a few hours to get a result in practice.
+
+> ### {% icon tip %} Tip: the running time of the program
+>
+> * Graph problem is way more complicated than we thought.
+> * It might take a much longer time for some complicated datasets.
+> * We can use multiple CPUs to accelerate the computation (for now, this function is not available in Galaxy yet),
+>   but, to use that, you can install Heinz directly via [Bioconda](https://anaconda.org/bioconda/heinz) in a Linux environment.
+{: .tip}
+
+## Visualize the output: visualize the optimal subnetwork
+
+The result we got from the last step is not very human readable. Therefore we need to visualize the output by making it
+into graphs. Except the tool we will use in Galaxy, you may consider using eXamine plugin in Cytoscape for a richer
+visualization.
+
+> ### {% icon hands_on %} Hands-on: visualize the optimal subnetwork
+>
+> - **Visualize** the optimal scoring subnetwork {% icon tool %} with the following parameters
+>   - *"Heinz output file"*: the output of **Identify optimal scoring subnetwork**
+>
+{: .hands_on}
+
```

Comment:
the tutorial ends a bit abruptly, can you maybe add a section about what we can deduce from this analysis/visualization, do we draw any conclusions? give it a bit of a wrap-up :)

Reply:
Fixed it!